### PR TITLE
feat: Added Avatars and profile pictures in userAutoComplete

### LINKED ISSUE #8869: Show user avatar in User Autocomplete and User list page
**Is your feature request related to a problem? Please describe.**
Currently the [`UserAutocompleteFormField`](https://github.com/ohcnetwork/care_fe/blob/develop/src/Components/Common/UserAutocompleteFormField.tsx#L82) and the Users list page does not show the avatar of the user.

**Describe the solution you'd like**
- [ ] Show the avatar of the user in each of the options for [`UserAutocompleteFormField`](https://github.com/ohcnetwork/care_fe/blob/develop/src/Components/Common/UserAutocompleteFormField.tsx#L82)
- [x] Show the avatar of the users in the Users list page ([here](https://github.com/ohcnetwork/care_fe/blob/develop/src/Components/Users/ManageUsers.tsx#L192))

**Additional context**
Image

Image

### PR DESCRIPTION
**Pull request by noufalrahim**
## Proposed Changes

- Fixes #8869 
- Added UserAvatar in usersList and user suggestion
- More?

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ 
Screenshot from 2024-10-23 11-39-54
Screenshot from 2024-10-23 11-40-58
] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/Components/Form/FormFields/Autocomplete.tsx [hunk 1]
```diff
@@ -24,6 +25,7 @@ type AutocompleteFormFieldProps<T, V> = FormFieldBaseProps<V> & {
   optionValue?: OptionCallback<T, V>;
   optionDescription?: OptionCallback<T, string>;
   optionIcon?: OptionCallback<T, React.ReactNode>;
+  optionProfilePicture?: any;
```
Comment (rithviknishad) [inline src/Components/Form/FormFields/Autocomplete.tsx]: let's not use `any` types

File: src/Components/Form/FormFields/Autocomplete.tsx [hunk 2]
```diff
@@ -74,6 +76,7 @@ type AutocompleteProps<T, V = T> = {
   placeholder?: string;
   optionLabel: OptionCallback<T, string>;
   optionIcon?: OptionCallback<T, React.ReactNode>;
+  optionProfilePicture?: any;
```
Comment (rithviknishad) [inline src/Components/Form/FormFields/Autocomplete.tsx]: `any` not allowed

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by bodhish** [comment]
image
This looks broken

**Comment by noufalrahim** [comment]
Sorry, I didn't understand..how is it broken..

File: src/Components/Form/FormFields/Autocomplete.tsx (hunk 1 above)
Comment (rithviknishad) [inline src/Components/Form/FormFields/Autocomplete.tsx]: Also, this is a reusable generic component. `optionProfilePicture` is too specific for a reusable generic component.

Comment (noufalrahim) [reply src/Components/Form/FormFields/Autocomplete.tsx]: Updated

**Comment by rithviknishad** [comment]
image

enlarging the image such that the `optionDescription` (i.e. 'Doctor - @rithviknishad-doctor') comes to the right of the image would be a better UI

**Comment by noufalrahim** [comment]
Screenshot from 2024-10-23 15-15-02
is this ok?

**Comment by rithviknishad** [comment]
Better. The online dot could come to the bottom right of the avatar. Also prefix `@` to the username

**Comment by noufalrahim** [comment]
Screenshot from 2024-10-23 15-25-36
How is this?

**Comment by rithviknishad** [comment]
Yup!

**Comment by noufalrahim** [comment]
Updated!

File: src/Components/Common/UserAutocompleteFormField.tsx [hunk 3]
```diff
@@ -80,6 +80,9 @@ export default function UserAutocomplete(props: UserSearchProps) {
         )}
         optionLabel={formatName}
         optionIcon={userOnlineDot}
+        optionImage={(option: UserBareMinimum | UserModel) =>
+          (option as UserModel).read_profile_picture_url || ""
+        }
```
Comment (rithviknishad) [inline src/Components/Common/UserAutocompleteFormField.tsx]: if `read_profile_picture_url` is not present in `UserBareMinimum` but backend provides it, then add then move the `read_profile_picture_url` to `UserBareMinimum` instead.

```suggestion
        optionImage={(option: UserBareMinimum) => option.read_profile_picture_url}
```

File: src/Components/Form/FormFields/Autocomplete.tsx [hunk 4]
```diff
@@ -236,24 +242,42 @@ export const Autocomplete = <T, V>(props: AutocompleteProps<T, V>) => {
...
+                              />
+                              <span className="absolute bottom-0 right-0 z-10">
+                                {option.icon}
+                              </span>
+                            </div>
+                          </div>
+                          <div className="w-full">
+                            <div className="flex w-full justify-between">
+                              <span>
+                                {"@"}
```
Comment (rithviknishad) [inline src/Components/Form/FormFields/Autocomplete.tsx]: @noufalrahim like said previously, this is a reusable component. Modifications made here will affect all other places. `@` prefix is only supposed to be present for `UserAutocompleteFormField` not any other usages.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Common/UserAutocompleteFormField.tsx (hunk 3 above)
Comment (noufalrahim) [reply src/Components/Common/UserAutocompleteFormField.tsx]: Done..

File: src/Components/Form/FormFields/Autocomplete.tsx (hunk 4 above)
Comment (noufalrahim) [reply src/Components/Form/FormFields/Autocomplete.tsx]: So where am I supposed to add "@"

Comment (noufalrahim) [reply src/Components/Form/FormFields/Autocomplete.tsx]: can I add it in utils formatName?
Screenshot from 2024-10-23 16-57-05

Comment (rithviknishad) [reply src/Components/Form/FormFields/Autocomplete.tsx]: The `username` should have the prefix `@`, not the name

Comment (rithviknishad) [reply src/Components/Form/FormFields/Autocomplete.tsx]: You can add it in `optionDescription` of the UserAutocompleteFormField right?

image

**Comment by noufalrahim** [comment]
I have a doubt regarding conflict.. How to resolve it..

**Comment by rithviknishad** [comment]
merge latest develop into this branch and clear conflicts from the editor and push the merge commit.

**Comment by noufalrahim** [comment]
Opened another PR

**Pull request closed without merging**